src/routes/auth/user.ts

```diff
@@ -35,28 +32,30 @@ userAuthRouter.post("/register", async (req, res) => {
   if (userUsernameDoc) return res.status(400).json({ username: "this username is already taken" });
   if (userEmailDoc) return res.status(400).json({ email: "this email is already taken" });
 
-  // Gen salt and store it together with the hashed password
-  const userPasswordSalt = await bcrypt.genSalt(10);
-  const hashedPassword = <string>await bcrypt.hash(newUser.password, userPasswordSalt).catch((err) => console.log(err));
-
-  // Add user to DB
-  const _newUserModel = (
-    await UserModel.create({
-      email: newUser.email,
-      username: newUser.username,
-      password: hashedPassword,
-      password_salt: userPasswordSalt,
-      private_profile: false,
-      banned: false,
-    })
-  ).save();
-
-  // Return auth token;
-  let accessToken;
-  const TOKEN = process.env.ACCESS_TOKEN_SECRET;
-  TOKEN ? (accessToken = jwt.sign({ email: newUser.email, username: newUser.username }, TOKEN)) : console.log("No secret access token in env");
-
-  res.status(200).json({ jwt: `Bearer ${accessToken}` });
+  try {
+    // Gen salt and store it together with the hashed password
+    const userPasswordSalt = await bcrypt.genSalt(10);
+    const hashedPassword = await bcrypt.hash(newUser.password, userPasswordSalt);
+
+    // Add user to DB
+    const _newUserModel = (
+      await UserModel.create({
+        email: newUser.email,
+        username: newUser.username,
+        password: hashedPassword,
+        password_salt: userPasswordSalt,
+        private_profile: false,
+        banned: false,
+      })
+    ).save();
+
+    // Return auth token;
+    const accessToken = jwt.sign({ email: newUser.email, username: newUser.username }, process.env.JWT_SECRET_USER);
+
+    res.status(200).json({ jwt: `Bearer ${accessToken}` });
+  } catch (err) {
+    console.log(err);
```

Comment:
Better to send an error to express (`next(error)`). You can use this to generate nice errors: https://www.npmjs.com/package/http-errors

Here it's also worse than before since you don't even res anything in case of error. Meaning if something goes wrong the request will never terminate.